trio/tests/test_signals.py

```diff
@@ -60,50 +134,44 @@ def test_catch_signals_wrong_thread():
     def direct_handler(signo, frame):
         delivered_directly.add(signo)
 
-    async def wait_call_soon_idempotent_queue_barrier():
-        ev = Event()
-        token = _core.current_trio_token()
-        token.run_sync_soon(ev.set, idempotent=True)
-        await ev.wait()
-
     print(1)
```

Comment:
Maybe print something more descriptive?

Reply:
Hmm, actually, have you considered just adding some branch logic for each 
`await wait_run_sync_soon_idempotent_queue_barrier()` location and then just use `pytest.mark.parametrize()` over signals and that?

Might reduce some code unless the duplication is intentional.

Reply:
This is a pretty complicated test that I didn't really touch here, so honestly I'm not sure what it does right now :-). I doubt it's worth messing with – if it breaks we can easily add better print statements or whatever then.